### PR TITLE
Add some base plugins and configuration

### PR DESCRIPTION
**Pull request by slackersoft**

### CODE REVIEW

File: vimrc [hunk 1]
```diff
@@ -68,6 +133,8 @@ endif
 "  endfor
 "endfor
 
+autocmd FocusLost * silent! wall
```
Comment (sgravrock) [inline vimrc]: Based on some googling, it looks like this automatically saves when vim loses focus. I could easily see that causing data loss if someone didn't know it was there and assumed default vim behavior.

Comment (slackersoft) [reply vimrc]: That is indeed the intent of that setting. I _think_ it really only works in a GUI though. I could see an argument to put it into the existing `if has("gui_running")` check or having that just need to be pulled out into my personal `.vimrc.local`. Thoughts?

File: vimrc [hunk 2]
```diff
@@ -30,6 +30,24 @@
...
+filetype off
+
+call plug#begin('~/.vim/plugged')
+
+Plug 'epmatsw/ag.vim'
+Plug 'scrooloose/nerdtree', { 'on':  'NERDTreeToggle' }
+Plug 'ctrlpvim/ctrlp.vim'
+Plug 'tpope/vim-surround'
+Plug 'tpope/vim-unimpaired'
+Plug 'editorconfig/editorconfig-vim'
```
Comment (kejadlen) [inline vimrc]: Looks like we already have editorconfig via a native vim package: https://github.com/seattle-beach/vim-config/tree/master/pack/foo/start/editorconfig

Comment (slackersoft) [reply vimrc]: cool, I think I verified I'm still getting editorconfig stuff after removing that.

File: vimrc (hunk 1 above)
Comment (sgravrock) [reply vimrc]: I think .vimrc.local would be the way to go for this one.

Comment (slackersoft) [reply vimrc]: done

**Comment by sgravrock** [comment]
The parts of this that I understand look good to me. If there are no pending objections from anyone else, I say merge it.

**Pull request merged**